cmds/kexec/kexec_linux.go

```diff
@@ -135,21 +141,72 @@ func main() {
 		}
 	}
 
+	// mbk indicates that we are a multiboot kernel
+	var mbk bool
+	var kernelpath string
 	if opts.load {
-		kernelpath := flag.Args()[0]
+		kernelpath = flag.Args()[0]
 		log.Printf("Loading %s for kernel\n", kernelpath)
 
-		var l loader = file{opts.initramfs}
 		if err := multiboot.Probe(kernelpath); err == nil {
+			mbk = true
+		} else if err == multiboot.ErrFlagsNotSupported {
+			log.Fatal(err)
+		}
+	}
+
+	var segs []kexec.Segment // used to add other segments for, e.g., multiboot
+	if opts.acpi != "" {
+		if !mbk {
+			log.Fatalf("You can only specify -a with multiboot kernels")
+		}
+		// NewSDT won't work on some linux kernels that limit reading
+		// above 1m. So we have to read the rsdp, which seems ok; then cons up
+		// a new SDT from scratch, since there is not one in /sys.
+		r, err := acpi.GetRSDP()
+		if err != nil {
+			log.Fatalf("Trying to read RSDP: %v", err)
+		}
+
+		s, err := acpi.NewSDT()
+		if err != nil {
+			log.Fatal(err)
+		}
+		s.Base = r.Base()
+		// We can't use the table pointers in SDT as Linux seems to want to deny
+		// access. So we have an SDT full of pointers we can't use. Awesome!
+		// And we can't read the SDT. I'm just chock full of good news today.
+		// acpi.RawTables reads from files in /sys
+		rr, err := acpi.RawTables()
+		if err != nil {
+			log.Fatal(err)
+		}
+		xtra, err := acpi.RawFromFile(opts.acpi)
+		if err != nil {
+			log.Fatal(err)
+		}
+		rr = append(rr, xtra)
+		log.Printf("Calling SDT Marshal with %d tables", rr)
+		b, err := s.MarshalAll(rr...)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Printf("Marshal'ed %d bytes for ACPI segment", len(b))
+		seg := kexec.NewSegment(b, kexec.Range{Start: uintptr(s.Base), Size: uint(len(b))})
```

Comment:
When you overwrite the ACPI, how do you know there's enough space to accommodate the larger size of `b`? Do you have to update the size of the ACPI range in e820 or anything like that?

Reply:
I'm gonna mark the segment now. This might even work.